### PR TITLE
Refactor PostProcessor to output result of mechanical simulation

### PR DESCRIPTION
**Pull request by Rombur**
Refactor `PostProcessor`  to be able to output thermal results, mechanical results, or thermo-mechanical results. For now, the mechanical output is limited to displacement and strain. Adding stress is not difficult but it requires to also pass the `MaterialProperty`. I've let that for later.

### CODE REVIEW

File: source/PostProcessor.cc [hunk 1]
```diff
@@ -15,13 +15,55 @@
...
 template <int dim>
 PostProcessor<dim>::PostProcessor(MPI_Comm const &communicator,
                                   boost::property_tree::ptree const &database,
                                   dealii::DoFHandler<dim> &dof_handler,
                                   int ensemble_member_index)
-    : _communicator(communicator), _dof_handler(dof_handler)
+    : _communicator(communicator)
 {
+  // This is an internal data we embded in the data base. We don't expose
+  // it to the user
```
Comment (stvdwtt) [inline source/PostProcessor.cc:54]: What does "internal" mean here? First I interpreted it to mean that the "thermal_output" and "mechanical_output" entries are set within the PostProcessor class, but that doesn't seem to be true. Does "internal" mean that it gets set in the application, not the input file?

File: source/PostProcessor.cc [hunk 2]
```diff
@@ -15,13 +15,55 @@
...
 namespace adamantine
 {
+template <int dim>
+StrainPostProcessor<dim>::StrainPostProcessor()
+    : dealii::DataPostprocessorTensor<dim>("strain", dealii::update_gradients)
+{
+}
+
+template <int dim>
+void StrainPostProcessor<dim>::evaluate_vector_field(
```
Comment (stvdwtt) [inline source/PostProcessor.cc:25]: The name makes it seem like this function is more general than it is. I'd recommend changing it to something like `calc_strain_from_displacement`.

Comment (Rombur) [reply source/PostProcessor.cc:25]: I don't have a choice in the name of the function. The class is derived from `DataPostprocessorTensor` and I need to override that function for deal.II to do the right thing

Comment (stvdwtt) [reply source/PostProcessor.cc:25]: Oh, I see, I didn't catch that the class was derived from a deal.II class.

File: source/PostProcessor.cc (hunk 1 above)
Comment (stvdwtt) [reply source/PostProcessor.cc:54]: I think using the property_tree is perfectly fine, I just wanted to make sure I understood the information flow. I have a suggested edit for the comment below.

[the remark this answers is not in the document]
Comment (Rombur) [reply source/PostProcessor.cc]: I have changed the comment

**Review by stvdwtt — APPROVED** [review]

**Pull request merged**